### PR TITLE
overwriting app name with fixed value

### PR DESCRIPTION
**Pull request by tnazarew**
Signed-off-by: tomasznazarewicz <t.nazarewicz94@gmail.com>

### Problem

A need to overwrite app name in JobName came up but funtionality was not implemented

Closes: #715 

### Solution

We decided to go with overwriting JobName with literal value given as an argument.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

### CODE REVIEW

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java [hunk 1]
```diff
@@ -223,11 +223,14 @@ protected OpenLineage.JobBuilder buildJob(QueryExecution queryExecution) {
     if (node instanceof WholeStageCodegenExec) {
       node = ((WholeStageCodegenExec) node).child();
     }
+
+
+    String name = eventEmitter.getOverwriteName().isPresent() ? eventEmitter.getOverwriteName().get() : sparkContext.appName();
```
Comment (mobuchowski) [inline integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java:228]: `eventEmitter.getOverwriteName().orElse(sparkContext.appName())`

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java [hunk 2]
```diff
@@ -20,13 +22,15 @@
 import org.apache.commons.lang3.StringUtils;
 import org.apache.hc.core5.http.NameValuePair;
 import org.apache.hc.core5.net.URLEncodedUtils;
+import org.jetbrains.annotations.NotNull;
```
Comment (mobuchowski) [inline integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java:25]: Wrong `@NotNull` 😉

**Review by mobuchowski — CHANGES_REQUESTED** [review]

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java [hunk 3]
```diff
@@ -312,16 +313,18 @@ private ArgumentParser parseConf(SparkConf conf) {
...
       Optional<Map<String, String>> urlParams =
           findSparkUrlParams(conf, SPARK_CONF_URL_PARAM_PREFIX);
 
       boolean consoleMode =
           findSparkConfigKey(conf, SPARK_CONF_CONSOLE_TRANSPORT)
-              .map(v -> Boolean.valueOf(v))
+              .map(Boolean::valueOf)
               .filter(v -> v)
               .orElse(false);
       return new ArgumentParser(
```
Comment (pawel-big-lebowski) [inline integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java:326]: Consider using builder for `ArgumenetParser`.

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java [hunk 4]
```diff
@@ -223,11 +223,14 @@ protected OpenLineage.JobBuilder buildJob(QueryExecution queryExecution) {
...
     return openLineage
         .newJobBuilder()
         .namespace(this.eventEmitter.getJobNamespace())
         .name(
-            sparkContext.appName().replaceAll(CAMEL_TO_SNAKE_CASE, "_$1").toLowerCase(Locale.ROOT)
```
Comment (pawel-big-lebowski) [inline integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java:230]: Consider extracting this to a method with some doc provided to describe what's happening here.

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java (hunk 1 above)
Comment (tnazarew) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java]: done

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java (hunk 2 above)
Comment (tnazarew) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java]: yeah it shouldn't be there, removed

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java (hunk 3 above)
Comment (tnazarew) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java]: done

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java (hunk 4 above)
Comment (tnazarew) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java:230]: done

**Review by pawel-big-lebowski — APPROVED** [review]
I think this is exactly what we wanted. 💯 🚀 🥇  
Minor comment added.

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/RddExecutionContext.java [hunk 5]
```diff
@@ -282,9 +282,14 @@ protected OpenLineage.Job buildJob(int jobId) {
     if (jobSuffix == null) {
       suffix = String.valueOf(jobId);
     }
-    String jobName = sparkContextOption.map(SparkContext::appName).orElse("unknown") + "." + suffix;
+
+    String name =
+        eventEmitter.getOverwriteName().isPresent()
```
[the remark this answers is not in the document]
Comment (tnazarew) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/RddExecutionContext.java:287]: yup

[the remark this answers is not in the document]
Comment (tnazarew) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/RddExecutionContext.java]: changed

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**